src/palimpzest/core/data/validationdata.py

```diff
@@ -0,0 +1,172 @@
+import json
+import os
+import logging
+from palimpzest.core.data.dataclasses import ExecutionStats
+
+logger = logging.getLogger(__name__)
+
+class ValidationData:
+    def __init__(self, file_path: str | None = None, annotations: dict | None = None):
+        """
+        Initialize ValidationData with a directory path containing execution_stats.updated.json.
+        Args:
+            file_path: Path to the execution stats file
+        """
+        self.file_path = file_path
+        self.annotations = annotations
+        self.expected_output = {}
+        self._input_dataset = None
+        
+        # Read execution stats JSON
+        if self.file_path is not None:
+            with open(self.file_path) as f:
+                execution_stats = json.load(f)
+        elif self.annotations is not None:
+            execution_stats = self.annotations
+        else:
+            execution_stats = ExecutionStats().to_json()
+
+        if "plan_stats" in execution_stats:
```

Comment:
Note to self: now that `execution_stats` has a separate field for `sentinel_plan_stats` we may want to process that as well?